### PR TITLE
feat(dashboard): Add relative step operators in step conditions fixes NV-6002

### PR DESCRIPTION
**Pull request by scopsy**
## Summary by CodeRabbit

- **New Features**
  - Added support for relative date operators (e.g., "more than X ago", "less than X ago", "within last", "not within last", "exactly X ago") in condition editors for date and datetime fields.
  - Introduced a dedicated editor for entering relative date values, allowing users to specify an amount and time unit.
  - Enhanced help text and usage examples for new relative date operators.

- **Bug Fixes**
  - Improved validation to ensure correct input and error handling for relative date conditions.

- **Tests**
  - Added comprehensive test coverage for all new relative date operators and edge cases.

### CODE REVIEW

**Comment by scopsy** [comment]
@coderabbitai summary

File: apps/api/src/app/shared/services/query-parser/query-parser.service.ts [hunk 1]
```diff
@@ -60,6 +70,78 @@ function validateBooleanInput(dataInput: unknown): BooleanValidation {
...
+    default:
+      // fallback to days if unit is not recognized
+      date.setDate(date.getDate() - amount);
+      break;
+  }
+
+  return date;
+}
+
+function getToleranceMs(unit: string): number {
```
Comment (scopsy) [inline apps/api/src/app/shared/services/query-parser/query-parser.service.ts:126]: Tolerance helps with not exact messages a less than 1 day  ago is pretty lose term.

File: apps/dashboard/src/components/conditions-editor/help-icon.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import type { HelpTextInfo } from '@/components/conditions-editor/field-type-editors';
+
+type HelpIconProps = {
+  hasError: boolean;
+  errorMessage?: string;
+  helpText?: HelpTextInfo | null;
+  contentWidth?: string;
+};
+
+export function HelpIcon({ hasError, errorMessage, helpText, contentWidth = 'w-[240px]' }: HelpIconProps) {
```
Comment (scopsy) [inline apps/dashboard/src/components/conditions-editor/help-icon.tsx:14]: Refactored

File: apps/dashboard/src/components/conditions-editor/value-editor.tsx [hunk 3]
```diff
@@ -28,113 +53,242 @@ export const ValueEditor = (props: ValueEditorProps) => {
     return null;
   }
 
-  // Get dynamic placeholder and help text
   const placeholder = getPlaceholder ? getPlaceholder(field, operator) : 'value';
   const helpText = getHelpText ? getHelpText(field, operator) : null;
+  const hasError = !!error;
 
-  // Combined icon component that shows error or info content
```
Comment (scopsy) [inline apps/dashboard/src/components/conditions-editor/value-editor.tsx:35]: This component was refactored to make it easier to maintain, and split to smaller cmponents

File: apps/dashboard/src/components/workflow-editor/steps/utils/preview-context-storage.utils.ts [hunk 4]
```diff
@@ -121,7 +121,7 @@ export function mergePreviewContextData(persistedData: ParsedData, serverDefault
   };
 }
 
-function mergeObjectData<T extends Record<string, unknown>>(persisted: T, serverDefault: T): T {
+export function mergeObjectData<T extends Record<string, unknown>>(persisted: T, serverDefault: T): T {
```
Comment (scopsy) [inline apps/dashboard/src/components/workflow-editor/steps/utils/preview-context-storage.utils.ts:124]: Not related to this PR, but found a bug with mergin payload example from the server when the user added new keys to the schema
